### PR TITLE
gh-93692: remove redundant "build finished successfully" message from setup.py

### LINKED ISSUE #93692: setup.py only reports successful builds if one or more stdlib modules are missing
_"Python build finished successfully!"_ is only printed if there were missing modules:

https://github.com/python/cpython/blob/cf730b595eea0460a7305205f7dfb6ecf0346351/setup.py#L539-L547

Either we print it every time the build succeeds:

```diff
diff --git a/setup.py b/setup.py
index 4c497346e8..e87da57a47 100644
--- a/setup.py
+++ b/setup.py
@@ -538,5 +538,9 @@ def print_three_column(lst):
 
-        if self.missing:
+        if not self.failed:
             print()
             print("Python build finished successfully!")
+            print()
+
+        if self.missing:
+            print()
             print("The necessary bits to build these optional modules were not "
```

Or we just remove that line from `setup.py`.

I'm fine with either.

_Originally reported by @izumiberat in https://github.com/python/cpython/issues/93659#issuecomment-1152435498_

### PR DESCRIPTION
**Pull request by erlend-aasland**
Resolves #93692 

* Issue: gh-93692

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -536,9 +536,13 @@ def print_three_column(lst):
                 print("%-*s   %-*s   %-*s" % (longest, e, longest, f,
                                               longest, g))
 
-        if self.missing:
+        if not self.failed:
```
Comment (AA-Turner) [inline setup.py]: The check at the end of the function (Line 599 before this commit) is `self.failed or self.failed_on_import or self.missing`. Perhaps `failed_on_import` should be included here?

```suggestion
        if not self.failed and not self.failed_on_import:
```

A

Comment (erlend-aasland) [reply setup.py]: Yeah... or maybe we should just remove this message. Thinking about this today, I lean towards just removing this `print`.

**Comment by erlend-aasland** [comment]
@tiran, are you fine with this? There is no point in printing this message only if some modules failed to build.

**Review by tiran — APPROVED** [review]
Yeah, makes sense :)

**Pull request merged**